lib/reply.js

```diff
@@ -203,25 +203,23 @@ function sendStream (payload, res, reply) {
 
 function handleError (reply, error, cb) {
   var statusCode = reply.res.statusCode
-  if (error == null) {
-    statusCode = statusCode || 500
-  } else if (error.status >= 400) {
-    if (error.status === 404) {
-      notFound(reply)
-      return
-    }
-    statusCode = error.status
-  } else if (error.statusCode >= 400) {
-    if (error.statusCode === 404) {
-      notFound(reply)
-      return
+  statusCode = (statusCode && statusCode >= 400) ? statusCode : 500
```

Comment:
Here you can just use `statusCode >= 400`, there is no need to perform the existency check.
If the value exists it should be a number, if not it will be evaluated as `0`.

Reply:
Done